config/cmake/addMPI.cmake

```diff
@@ -0,0 +1,44 @@
+# LLNS Copyright Start
+# Copyright (c) 2014-2018, Lawrence Livermore National Security
+# This work was performed under the auspices of the U.S. Department
+# of Energy by Lawrence Livermore National Laboratory in part under
+# Contract W-7405-Eng-48 and in part under Contract DE-AC52-07NA27344.
+# Produced at the Lawrence Livermore National Laboratory.
+# All rights reserved.
+# For details, see the LICENSE file.
+# LLNS Copyright End
+
+
+##############################################################################
+# Copyright � 2017-2018,
+# Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable Energy, LLC
+#All rights reserved. See LICENSE file and DISCLAIMER for more details.
+##############################################################################
+
+if(MSVC)
+   #This is for finding MS-MPI.
+   #set(_MPI_PREFIX_PATH)
+  #list(APPEND _MPI_PREFIX_PATH
+ #"[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\MPI;Path]/..")
+  # set(MPIEXEC "C:/Program Files/Microsoft MPI/Bin/mpiexec.exe")
+   # For building MPI programs the selected Visual Studio compiler is used,
+ #namely cl.exe.
+  # So there is no need to set a specific MPI compiler.
+  set(MPI_CXX_COMPILER "${CMAKE_CXX_COMPILER}")
+  set(MPI_C_COMPILER "${CMAKE_C_COMPILER}")
+  set(MPI_CXX_INCLUDE_PATH "C:/Program Files (x86)/Microsoft SDKs/MPI/Include")
+   # Make sure the correct libraries (64-bit or 32-bit) are selected.
+   # Decide between 32-bit and 64-bit libraries for Microsoft's MPI
+   if("${CMAKE_SIZEOF_VOID_P}" EQUAL 8)
+     set(MS_MPI_ARCH_DIR x64)
+  else()
+    set(MS_MPI_ARCH_DIR x86)
+   endif()
+  set(MPI_CXX_LIBRARIES "C:/Program Files (x86)/Microsoft SDKs/MPI/Lib/${MS_MPI_ARCH_DIR}/msmpi.lib")
+   set(MPI_C_INCLUDE_PATH "${MPI_CXX_INCLUDE_PATH}")
+   set(MPI_C_LIBRARIES "{${MPI_CXX_LIBRARIES}")
+	set(MPI_C_FOUND YES)
+	set(MPI_CXX_FOUND YES)
+ else()
+    find_package(MPI)
```

Comment:
It looks like find_package should work for detecting MSMPI and MPICH2 installs on Windows -- was it not working on your system?

Reply:
No, it found the mpiexec but didn't find the header files, or get appropriate compilers.  I don't think this is sufficient but the find_package(MPI) didn't seem to work very well on windows.  I am going to play with it some more as we go along.  Not entirely sure what is in place for windows was appropriate either but it got further along than just with the find_package

Reply:
Maybe a problem with the MSMPI installer not always setting environment variables. What about calling find_package(MPI) first for all systems, and then as a fallback if things aren't found, use the hard-coded default paths?

Reply:
I updated it to use  the findMPI,,  which is still a bit finicky on Windows.  Sometimes it works, other times it doesn't so well.